web/packages/teleterm/src/ui/DeepLinks/launchDeepLink.ts

```diff
@@ -0,0 +1,266 @@
+
+  if (!result.isAtDesiredWorkspace) {
+    return;
+  }
+
+  const { rootClusterUri } = result;
+
+  workspacesService
+    .getWorkspaceDocumentService(rootClusterUri)
+    .openConnectMyComputerDocument({ rootClusterUri });
+}
+
+/**
+ * loginAndSetActiveWorkspace will set the relevant cluster if it is in the app and, if not,
+ * it opens a login dialog with cluster address and username prefilled from the URL.
+ */
+async function loginAndSetActiveWorkspace(
+  { clustersService, modalsService, workspacesService }: IAppContext,
+  url: DeepURL
+): Promise<
+  | {
+      isAtDesiredWorkspace: false;
+    }
+  | {
+      isAtDesiredWorkspace: true;
+      rootClusterUri: RootClusterUri;
+    }
+> {
+  const currentlyActiveWorkspace = workspacesService.getRootClusterUri();
+  // If we closed the dialog to reopen documents when launching a deep link,
+  // setting the active workspace again will reopen it.
+  const reopenCurrentlyActiveWorkspace = async () => {
+    if (currentlyActiveWorkspace) {
+      await workspacesService.setActiveWorkspace(currentlyActiveWorkspace);
+    }
+  };
+  const rootClusterId = url.hostname;
+  const clusterAddress = url.host;
+  const prefill = {
+    clusterAddress,
+    username: url.username,
+  };
+
+  const rootClusterUri = routing.getClusterUri({ rootClusterId });
+  const rootCluster = clustersService.findCluster(rootClusterUri);
+
+  if (!rootCluster) {
+    const { canceled } = await new Promise<{ canceled: boolean }>(resolve => {
+      modalsService.openRegularDialog({
+        kind: 'cluster-connect',
+        clusterUri: undefined,
+        reason: undefined,
+        prefill,
+        onSuccess: () => resolve({ canceled: false }),
+        onCancel: () => resolve({ canceled: true }),
+      });
+    });
+
+    if (canceled) {
+      await reopenCurrentlyActiveWorkspace();
+      return {
+        isAtDesiredWorkspace: false,
+      };
+    }
+  }
+
+  const { isAtDesiredWorkspace } = await workspacesService.setActiveWorkspace(
+    rootClusterUri,
+    // prefill has to be passed here as wellin case the cluster is in the state (so the call
+    // to open cluster-connect above will be skipped) but there's no active cert. In that
+    // case, WorkspacesService will open cluster-connect itself with just the login step.
+    prefill
+  );
+
+  if (isAtDesiredWorkspace) {
+    return { isAtDesiredWorkspace: true, rootClusterUri };
+  }
+
+  await reopenCurrentlyActiveWorkspace();
+  return { isAtDesiredWorkspace: false };
+}
+
+async function openVnetPanel(
+  appCtx: IAppContext,
+  { vnet, connections }: AuxContext,
+  url: VnetDeepURL
+): Promise<void> {
+  const { notificationsService } = appCtx;
+  if (!vnet.isSupported) {
+    notificationsService.notifyWarning(
+      'VNet is not supported on this platform.'
+    );
+    return;
+  }
+
+  const result = await loginAndSetActiveWorkspace(appCtx, url);
+
+  if (!result.isAtDesiredWorkspace) {
+    return;
+  }
+
+  connections.open('vnet');
```

Comment:
When I was testing it, I realized that it may not be obvious how to start VNet in this panel (especially for the first time users):
<img width="403" alt="image" src="https://github.com/user-attachments/assets/713d6b7a-c8d2-41e8-9d63-2de2003876ca" />

Something like this could be clearer (but it’s just a suggestion):
<img width="403" alt="image" src="https://github.com/user-attachments/assets/c882b675-2970-4742-82c9-b5f5a4ae3411" />

Reply:
Yeah it's kinda bad. I'll see what I can do about it.

BTW, in this case the icon would be before the button text, citing Kenny here on whether the external link icon should be before or after the text:

> After. We use a leading icon to indicate where you're going, but this icon is really about how you're going rather than the destination/page itself

Reply:
This looks much better I think.

The stop button uses `fill="minimal"` on purpose. If it had any other fill, it'd look too different compared to the other buttons in that row and the "Open Diag Report" button.

https://github.com/user-attachments/assets/6695c8b6-2ea6-4cf2-8d5c-4f12be0f905d